Server/Server.h

```diff
@@ -3,24 +3,16 @@
 #include "gmock/gmock.h"
 
 namespace Server {
-	const int EXPECTED_FOO = 3;
-	int getFOO();
-
-	__interface IServer
-	{
-		Communication::IClientRequest& ReceiveRequest();
-	};
-
-	class Server
+	class Server : public Communicators::Responder
```

Comment:
The Server class implements the Communicators::Responder interface as well as contains a Communicators::Responder object inside of it. It's a decorator pattern :)